src/location.ts

```diff
@@ -91,15 +91,14 @@ export class Locator {
 
     let end = getPosition(endNeedle, this.#map);
 
-    // e.g. tsc that doesnt include } in source maps
-    if (end === null) {
-      endNeedle.column++;
-      end = getPosition(endNeedle, this.#map);
-    }
-
     if (end === null) {
+      // search the previous lines as the mapping was not found on the same line
+      // e.g. tsc that doesnt include } in source maps
+      for (let line = endNeedle.line; line > 0 && end === null; line--) {
+        end = getPosition({ line, column: Number.MAX_SAFE_INTEGER }, this.#map);
```

Comment:
Does `Number.MAX_SAFE_INTEGER` have some special meaning here or could this be `Infinity` too? It's just to search the end of the line?

Reply:
`Number.MAX_SAFE_INTEGER` was the one that came up in my head first. I'll replace it with `Infinity`. It would have a small overhead because of float & integer mix, but I guess it would be negligible.